src/utils/utils.py

```
@@ -0,0 +1,108 @@
+    copy_source = { 'Bucket': s3_bucket, 'Key': trainigs_model_path }
+    s3.meta.client.copy(copy_source, s3_bucket, 'otherkey')
+
+
+def read_config(data_type):
+    with open(f'./{data_type}/{data_type}_model_config.yaml', 'r') as f:
+        conf = yaml.safe_load(f)
+
+    return conf
+
+
+def load_json_file(local_path):
+    with open(local_path, 'r') as f:
+        return json.load(f)
+
+
+def load_yaml_file(local_path):
+    with open(local_path, 'r') as f:
+        return yaml.safe_load(f)
+
+
+def save_text(text, local_path):
+    with open(local_path, 'w+') as f:
+        f.write(text)
+
+
+def set_validated_model_to_current(model_datestamp, conf):
+     log.info("Setting the validated model %s to become current", model_datestamp)
+     model_path_trainings = f"s3://{conf['s3_bucket']}/{conf['s3']['trainings_model_path']}/{model_datestamp}"
+     model_path_current = f"s3://{conf['s3_bucket']}/{conf['s3']['current_model_path']}"
+
+     cmd = f'aws s3 cp {model_path_trainings} {model_path_current} --recursive'
+     os.system(cmd)
+
+
+def get_model_data_from_s3(conf, model_datestamp=None):
+    """Copies all model data given path (i.e model, meta, config and labels)
+    :param conf: model config for path definitions
+    :return: a MobileNetV2 model object
+    """
+    try:
+        local_model_dir = conf['local']['model_dir']
+        if model_datestamp:
+            model_path = f"s3://{conf['s3_bucket']}/{conf['s3']['trainings_model_path']}/{model_datestamp}"
+        else:
+            model_path = f"s3://{conf['s3_bucket']}/{conf['s3']['current_model_path']}"
+        log.info('Coping model data from %s to %s', model_path, local_model_dir)
+
+        cmd_cleanup = f'rm -rf {local_model_dir}'
+        cmd = f'aws s3 cp {model_path} {local_model_dir} --recursive'
+        os.system(cmd_cleanup)
+        os.system(cmd)
+
+    except Exception as e:
+        log.error('Model wasn\'t loaded, error: %s', e)
+        exit(1)
```

`exit(1)` seems pretty extreme. Why not just reraise the exception?

---

We can also re-raise, but either way wanted to **halt** the execution in this case.

---

My preference is to keep program logic like this as close to the actual program. Crashing the program from a dependent library function seems counterintuitive to me.